core/src/wallet/tezos/explorers/TezosLikeBlockchainExplorer.cpp

```diff
@@ -158,5 +158,27 @@ namespace ledger {
                         return false;
                     });
         }
+
+        Future<std:string> TezosLikeBlockchainExplorer::getCurrentDelegate(const std::string &address,
+                                                                           const std::shared_ptr<api::ExecutionContext> &context,
+                                                                           const std::shared_ptr<HttpClient> &http,
+                                                                           const std::string &rpcNode) {
+            const bool parseNumbersAsString = true;
+
+            std::unordered_map<std::string, std::string> headers{{"Content-Type", "application/json"}};
+            return http->GET(fmt::format("/chains/main/blocks/head/context/contracts/{}/delegate", address),
+                             std::unordered_map<std::string, std::string>{},
+                             rpcNode)
+                    .json(parseNumbersAsString)
+                    .map<std::string>(context, [](const HttpRequest::JsonResult &result) {
+                        auto &json = *std::get<1>(result);
+                        if (!json.IsString()) {
+                            return "";
+                        }
+                        return json.GetString();
+                    }).recover(context, [] (const Exception &exception) {
+                        return "";
```

Comment:
How to make difference between network issue for example and adresses without delegate ?